megatron/training.py

```diff
@@ -234,6 +277,8 @@ def get_model(neox_args, use_cache=False):
     print_rank_0("building GPT2 model ...")
 
     # Build model on cpu.
+    old_use_mup = neox_args.use_mup
```

Comment:
I assume the point of old_use_mup logic is to ensure we aren't using mup optimizers for the model? Add a comment to that effect if so. Explain why we use this logic if not.

Reply:
I added a comment explaining the original reason I added the old_use_mup logic. After looking at the docs again, I need to add another function that will reinitialize the model's weights using mup's functions (since setting use_mup to false here will make the model initialize with torch.nn's functions). Then, call that function after the weights were reinitialized.

Reply:
We discussed this offline. @nsarka is going to add a function to reinit all model weights if mup is enabled after running `set_base_shapes()`. The expected implementation method will be a for-loop over layers modeled after `to_sequential()`